src/Commands/EnvCreateCommand.php

```diff
@@ -150,18 +150,24 @@ public function createBuildEnv(
             $site_name = $site->getName();
             $project = $this->projectFromRemoteUrl($metadata['url']);
             $dashboard_url = "https://dashboard.pantheon.io/sites/{$site_id}#{$multidev}";
+            $pr_id = '';
+            if (preg_match('#^pr-([\d]+)$#', $multidev, $matches)) {
+                $pr_id = $matches[1];
+            }
+
             $metadata += [
                 'project' => $project,
                 'site-id' => $site_id,
                 'site' => $site_name,
                 'env' => $multidev,
+                'pr_id' => $pr_id,
                 'label' => $env_label,
                 'dashboard-url' => $dashboard_url,
                 'site-url' => "https://{$multidev}-{$site_name}.pantheonsite.io/",
                 'message' => "Created multidev environment [{$multidev}]({$dashboard_url}) for {$site_name}."
             ];
 
-            $command = $this->interpolate('terminus build:comment:add:commit --sha [[sha]] --message [[message]] --site_url [[site-url]]', $metadata);
```

Comment:
It seems to me that we should not eliminate the existing functionality of attaching a commit to an sha. It would be better if `build:commenet:add:commit` were used when the PR number is not known, and the new command used when the PR number is available.

Reply:
> It would be better if `build:commenet:add:commit` were used when the PR number is not known, and the new command used when the PR number is available.

Is it expected behavior to magically switch between the two? I think it'd better to be an explicit decision.

Reply:
If your site is set up per the default configuration, then the pull request number will always be available. If a build runs before a PR is available, though, then what is the fallback behavior? I'd recommend thinking of comments on commits in that way, as fallback behavior, rather than as an alternative.

Reply:
What about:

1. If `--pr-id=<num>` is specified as an argument, then we add a comment on the PR.
2. If no `--pr-id=<num>` is specified, then we add a comment on the changeset.

In this scenario, it'd be the responsibility of the calling script to indicate whether the comment is expected on the pull request vs. the changeset.

Reply:
> In this scenario, it'd be the responsibility of the calling script to indicate whether the comment is expected on the pull request vs. the changeset.

... which is what I think you were hinting at in your comment below.

Reply:
Yes, I think that you should add the comment to the commit if the PR number is not available.

Whether you get the PR via the current method proposed in this PR, or one of the alternative ways that I suggested, depends on which is most reliable. If all of the three providers we're supporting today have environment variables with the PR number (I believe they all do), and if there are no barriers to passing the PR number in via an option, then I'd recommend you go that way. Otherwise, I'm okay with sticking with the regex-based PR number extraction. The multidev name might conceivably be renamed to something other than PR-nnn (if the user customized their scripts), so I'd guess that using the environment variables would probably be more reliable.